text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,695 @@
+
+
+##### Field `start_line`
+
+Line number in source file.
+</details>
+
+
+<!-- messages -->
+
+### Example Payloads
+
+#### Simple example
+
+Considering the following example presented in a modified collapsed format:
+```
+foo;bar;baz 100 region=us,trace_id=0x01020304010203040102030401020304,span_id=0x9999999999999999 1687841528000000
+foo;bar 200 region=us
+```
+
+It represents 2 samples:
+* one for stacktrace `foo;bar;baz` with value `100`, attributes `region=us`, linked to trace `0x01020304010203040102030401020304` and span `0x9999999999999999`, and timestamp `1687841528000000`
+* one for stacktrace `foo;bar` with value `200`, attributes `region=us`, no link, no timestamp
+
+The resulting profile in OTLP format would look like this (in YAML format):
+```
+resource_profiles:
+  - resource:
+      attributes: null
+    schema_url: todo
+    scope_profiles:
+      - profiles:
+          - attribute_sets:
+              - attributes: null
+              - attributes:
+                  - key: region
+                    value:
+                      Value:
+                        string_value: us
+            functions:
+              - name_index: 1
+              - name_index: 2
+              - name_index: 3
+            links:
+              - span_id: ""
+                trace_id: ""
+              - span_id: "9999999999999999"
+                trace_id: "01020304010203040102030401020304"
+            locations:
+              - line:
+                  - {}
+              - line:
+                  - function_index: 1
+              - line:
+                  - function_index: 2
+            profile_types:
+              - attribute_set_indices:
+                  - 1
+                  - 1
+                link_indices:
+                  - 1
+                  - 0
+                stacktrace_indices:
+                  - 0
+                  - 1
+                timestamps:
+                  - 1.687841528e+15
+                  - 0
+                values:
+                  - 100
+                  - 200
+            stacktraces:
+              - location_indices:
+                  - 0
+                  - 1
+                  - 2
+              - location_indices:
+                  - 0
+                  - 1
+            string_table:
+              - ""
+              - foo
+              - bar
+              - baz
+            attributes: null
+            profile_id: 0102030405060708090a0b0c0d0e0f10
+        scope:
+          attributes: null
+```
+
+
+### Notable differences compared to other signals
+
+Due to the increased performance requirements associated with profiles signal, here are some notable differences between profiles signal and other signals.
+
+#### Relationships between messages
+
+There are two main ways relationships between messages are represented:
+* by embedding a message into another message (standard protobuf way)
+* by referencing a message by index (similar to how it's done in pprof)
```

Comment:
I actually suspect that table-lookup is a technique we MAY want to apply to other signals for significant reduction in size.   However, given the state of OTLP-Arrow, it could be a moot point.

IF we see successful implementations of profiling "processors" in the otel collector, I'd suggest we think about adding this capability to other Signals.

Reply:
We need to be careful with this. In my experiments I have seen in some cases for compressed payloads string lookup tables result in a regression (bigger compressed payloads).